variables.sh

```diff
@@ -64,3 +64,41 @@ END
 }
 
 export -f generate_dummy_coverage_file
+
+# pick_subset
+# $1: " " seperated string
+# $2: result var
+# $3: subset num
+# $4: subset total
+# e.g.
+#  $ pick_subset "a b c d" result 1 3
+#  $ echo $result
+#  a d
+function pick_subset()
```

Comment:
pick_subset takes a space separated string and breaks it up into multiple subsets. The first argument is the space separated string, the second argument is the name of the variable to store the output in, the third variable is the subset index and the fourth variable is the desired size of each subset.

```
# pick_subset "a b c d" result 0 3
# echo $result
#  c

# pick_subset "a b c d" result 1 3
# echo $result
#  a d

# pick_subset "a b c d" result 2 3
# echo $result
#  b

Note that the generated subsets do not necessarily get grouped in the same order as the original list (as seen in the example above)

Reply:
done